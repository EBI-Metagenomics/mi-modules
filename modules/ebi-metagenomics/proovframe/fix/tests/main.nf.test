nextflow_process {

    name "Test Process PROOVFRAME_FIX"
    script "../main.nf"
    process "PROOVFRAME_FIX"

    tag "modules"
    tag "modules_ebimetagenomics"
    tag "proovframe"
    tag "proovframe/fix"
    tag "proovframe/map"

    test("test - fa_tsv"){
        setup {
            run("PROOVFRAME_MAP") {
                script "../../map/main.nf"
                process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/emales.faa", checkIfExists: true),
                    file("${moduleDir}/tests/fixtures/emales-reads-sim.fa", checkIfExists: true),
                ]
                """
            }
            }
        }

        when {
            process {
                """                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file("${moduleDir}/tests/fixtures/emales.faa", checkIfExists: true),
                    file("${moduleDir}/tests/fixtures/emales-reads-sim.fa", checkIfExists: true),                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
                // See https://nf-co.re/docs/contributing/tutorials/nf-test_assertions for more information and examples.
            )
        }

    }

    // TODO nf-core: Change the test name preferably indicating the test-data and file-format used but keep the " - stub" suffix.
    test("sarscov2 - bam - stub") {

        options "-stub"

        when {
            process {
                """
                // TODO nf-core: define inputs of the process here. Example:
                
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.sorted.bam', checkIfExists: true),
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
                //TODO nf-core: Add all required assertions to verify the test output.
            )
        }

    }

}
