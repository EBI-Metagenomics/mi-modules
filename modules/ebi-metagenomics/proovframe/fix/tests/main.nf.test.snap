{
    "test - fa_tsv": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fa:md5,3e35d8421d3b8f2a1d9ebd46495b8141"
                    ]
                ],
                "1": [
                    "versions.yml:md5,285ddd6be0ed62527fe62fde1df82bf4"
                ],
                "out_fa": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fa:md5,3e35d8421d3b8f2a1d9ebd46495b8141"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,285ddd6be0ed62527fe62fde1df82bf4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.04.5"
        },
        "timestamp": "2024-10-18T09:15:05.574270365"
    },
    "test - faa_fasta - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,285ddd6be0ed62527fe62fde1df82bf4"
                ],
                "out_fa": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.fa:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,285ddd6be0ed62527fe62fde1df82bf4"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "23.04.5"
        },
        "timestamp": "2024-10-18T09:15:09.244102295"
    }
}